.cloud-build/notebook-execution-test-cloudbuild-single.yaml

```diff
@@ -11,20 +11,18 @@ steps:
     args:
     - -c
     - 'python3 .cloud-build/CheckPythonVersion.py'
-  # Upgrade pip
-  - name: ${_PYTHON_IMAGE}
-    entrypoint: pip
-    args: ['install', '--upgrade', 'pip']
   # Install Python dependencies
   - name: ${_PYTHON_IMAGE}
-    entrypoint: pip
-    args: ['install', '--upgrade',  '--user', '--requirement', '.cloud-build/requirements.txt']
+    entrypoint: /bin/sh
+    args:
+    - -c
+    - 'python3 -m pip install -U pip && python3 -m pip install -U --user -r .cloud-build/requirements.txt'
```

Comment:
I wonder if this would be a bit simpler and easier to read if you split this into two yaml actions (instead of using the shell && syntax). The shell syntax ensures the second part won't execute if the first one fails, but I would expect the same of the Cloud Build sequence.

Reply:
I originally had it that way (if you look at the first commit). The issue is that each build step starts fresh with the base image, and so the pip upgrade isn't persisted to future steps.

Reply:
ah, I see. Weird though, that every step is stateless. I prefer good old fashioned makefiles. :)